test/e2e/framework.go

```diff
@@ -1577,7 +1578,39 @@ func (data *TestData) PodWaitFor(timeout time.Duration, name, namespace string,
 // until the provided timeout expires).
 func (data *TestData) podWaitForRunning(timeout time.Duration, name, namespace string) error {
 	_, err := data.PodWaitFor(timeout, name, namespace, func(pod *corev1.Pod) (bool, error) {
-		return pod.Status.Phase == corev1.PodRunning, nil
+		if pod.Status.Phase != corev1.PodRunning {
+			return false, nil
+		} else {
+			node, err := data.clientset.CoreV1().Nodes().Get(context.TODO(), pod.Spec.NodeName, metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("error when getting Node for Pod '%s': %v", name, err)
+			}
+
+			osType := node.Status.NodeInfo.OperatingSystem
+			runtime := node.Status.NodeInfo.ContainerRuntimeVersion
+			if osType == "windows" && strings.Contains(runtime, "containerd") {
+				var stdout string
+				antctlCmd := fmt.Sprintf("C:/k/antrea/bin/antctl.exe get podinterface %s -n %s -o json", name, namespace)
+				envCmd := fmt.Sprintf("export POD_NAME=antrea-agent;export KUBERNETES_SERVICE_HOST=%s;export KUBERNETES_SERVICE_PORT=%d", clusterInfo.k8sServiceHost, clusterInfo.k8sServicePort)
+				cmd := fmt.Sprintf("%s && %s", envCmd, antctlCmd)
+				_, stdout, _, err = data.RunCommandOnNode(pod.Spec.NodeName, cmd)
```

Comment:
Question: can we run antctl commands inside antrea-agent Windows Pods with containerd runtime? If yes, I think we can have a task to change the antctl commands in e2e into container, then some logics are consistent with Linux? @XinShuYang

Reply:
Yes, I created an issue to track this task. Thanks for the suggestion! @wenyingd  https://github.com/antrea-io/antrea/issues/5612

Reply:
Yes, we should really try to avoid using `RunCommandOnNode` as much as possible in new e2e tests.